### PR TITLE
Upgrade StaderOracle implementation

### PR DESCRIPTION
**Pull request by blockgroot**

### CODE REVIEW

File: contracts/StaderOracle.sol [hunk 1]
```diff
@@ -119,6 +119,9 @@ contract StaderOracle is IStaderOracle, AccessControlUpgradeable, PausableUpgrad
         if (block.number < lastTrustedNodeCountChangeBlock + trustedNodeChangeCoolingPeriod) {
             revert CooldownNotComplete();
         }
+        if (trustedNodesCount == MIN_TRUSTED_NODES) {
```
Comment (evercoinx) [inline contracts/StaderOracle.sol:122]: Maybe to be more on the safe side and to make this check even broader like
`if (trustesNodesCount <= MIN_TRUSTED_NODES)`

Comment (blockgroot) [reply contracts/StaderOracle.sol:122]: `trustedNodesCount == MIN_TRUSTED_NODES` with this check we will never end up in a situation where trustedNodesCount < MIN_TRUSTED_NODES, and this is pre check before state gets update, so I think this check can serve our purpose.

Comment (blockgroot) [reply contracts/StaderOracle.sol]: [11c68d0](https://github.com/stader-labs/ethx/pull/258/commits/11c68d04b5950000c5428b209a3cc78cc7371ab4)

**Review by evercoinx — APPROVED** [review]

**Comment by manoj9april** [comment]
codeCov is really nice 🤟

File: contracts/StaderOracle.sol [hunk 2]
```diff
@@ -29,7 +29,7 @@ contract StaderOracle is IStaderOracle, AccessControlUpgradeable, PausableUpgrad
     uint256 public constant MAX_ER_UPDATE_FREQUENCY = 7200 * 7; // 7 days
     uint256 public constant ER_CHANGE_MAX_BPS = 10_000;
     uint256 public override erChangeLimit;
-    uint256 public constant MIN_TRUSTED_NODES = 5;
```
Comment (manoj9april) [inline contracts/StaderOracle.sol:32]: is this affecting any other usecase ? Do we require 5 nodes atleast on any usecase here ?

Comment (blockgroot) [reply contracts/StaderOracle.sol:32]: No, we decide to reduce the number of ONO's to the 3, and this strict check is preventing us to do so.

**Review by manoj9april — APPROVED** [review]
Looks good to me

File: contracts/StaderOracle.sol (hunk 2 above)
Comment (manoj9april) [reply contracts/StaderOracle.sol:32]: cool

File: contracts/StaderOracle.sol [hunk 3]
```diff
@@ -119,6 +119,9 @@ contract StaderOracle is IStaderOracle, AccessControlUpgradeable, PausableUpgrad
         if (block.number < lastTrustedNodeCountChangeBlock + trustedNodeChangeCoolingPeriod) {
             revert CooldownNotComplete();
         }
+        if (trustedNodesCount <= MIN_TRUSTED_NODES) {
```
Comment (galacticminter) [inline contracts/StaderOracle.sol:122]: Make the threshold for SD price update logic similar to every other update please.

Comment (blockgroot) [reply contracts/StaderOracle.sol:122]: [2d6a43f](https://github.com/stader-labs/ethx/pull/258/commits/2d6a43ff9e077c383e0583fef64b870d9d9c2520)

**Review by galacticminter — APPROVED** [review]

**Pull request merged**